src/controllers/courseController.ts

```diff
@@ -109,13 +108,10 @@ export class CourseController {
 				request.session!.save(() => {
 					response.redirect('/content-management/courses/title')
 				})
+			} else if (request.params.courseId) {
+				await this.editCourse(request, response)
+				response.redirect(`/content-management/courses/${request.params.courseId}/preview`)
```

Comment:
it doesn't need to return, as the flow is controlled with `if/else` now rather than `return`. They are separate routes in a sense, one's got course ID and one not.

Reply:
We could write the entire with one method and a complicated regex in the route if we wanted to but that doesn't make it desirable. 

Why not try to keep the code branches to a minimum in order to make it easier to test and understand and so reduce bugs?